python/cudf/cudf/core/reshape.py

```diff
@@ -57,6 +58,12 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     -------
     A new object of like type with rows from each object in ``objs``.
     """
+    
+    ''' Check if *objs* index contains any duplicates '''
+    for o in objs:
+        if not len(o.index) == len(set(o.index)):
+            raise ValueError("cannot reindex from a duplicate axis")
```

Comment:
`set(o.index)` will convert the GPU object to a host Python object element by element, so we definitely don't want to do that. There's an `nunique` attribute that we can use to get the number of unique elements to compare against the len of the object itself.

Reply:
Should this check run only if `axis=1`?

Reply:
There's also an edge case here where if the index of the `objs` are equals then it bypasses this check.